Functions/Admin/Set-RsEmailConfiguration.ps1

```diff
@@ -0,0 +1,175 @@
+            In order to maintain backwards compatibility, aliases with their respective name were created.
+            These then affect the default value for the new 'Authentication' parameter, which was introduced in this function.
+            The result should be, that all scripts using the old functions should remain functional, as the same effective outcome with the old call should persist.
+    #>
+    
+    [cmdletbinding()]
+    param
+    (
+        [ReportingServicesTools.SmtpAuthentication]
+        $Authentication = "Ntlm",
+        
+        [Parameter(Mandatory = $True)]
+        [string]
+        $SmtpServer,
+        
+        [Parameter(Mandatory = $True)]
+        [string]
+        $SenderAddress,
+        
+        [Alias('EmailCredentials')]
+        [System.Management.Automation.PSCredential]
+        $EmailCredential,
+        
+        [Alias('SqlServerInstance')]
+        [string]
+        $ReportServerInstance,
+        
+        [Alias('SqlServerVersion')]
+        [ReportingServicesTools.SqlServerVersion]
+        $ReportServerVersion,
+        
+        [string]
+        $ComputerName,
+        
+        [System.Management.Automation.PSCredential]
+        $Credential
+    )
+    
+    #region Connect to Report Server using WMI
+    try
+    {
+        $splat = @{ }
+        if ($PSBoundParameters.ContainsKey('ReportServerInstance')) { $splat['ReportServerInstance'] = $ReportServerInstance }
+        if ($PSBoundParameters.ContainsKey('ReportServerVersion')) { $splat['ReportServerVersion'] = $ReportServerVersion }
+        if ($PSBoundParameters.ContainsKey('ComputerName')) { $splat['ComputerName'] = $ComputerName }
+        if ($PSBoundParameters.ContainsKey('Credential')) { $splat['Credential'] = $Credential }
+        $rsWmiObject = New-RsConfigurationSettingObject @splat
+    }
+    catch
+    {
+        throw
+    }
+    #endregion Connect to Report Server using WMI
+    
+    #region Backwards compatibility
+    if (-not $PSBoundParameters.ContainsKey("Authentication"))
+    {
+        $CallName = (Get-PSCallStack)[0].InvocationInfo.InvocationName
+        
+        if ($CallName -eq "Set-RsEmailSettingsAsNoAuth") { $Authentication = "None" }
+        if ($CallName -eq "Set-RsEmailSettingsAsBasicAuth") { $Authentication = "Basic" }
+        if ($CallName -eq "Set-RsEmailSettingsAsNTLMAuth") { $Authentication = "Ntlm" }
+    }
+    #endregion Backwards compatibility
+    
+    if (($Authentication -like "Basic") -and (-not $EmailCredential))
+    {
+        throw (New-Object System.Management.Automation.PSArgumentException("Basic authentication requires passing credentials using the 'EmailCredential' parameter!"))
+    }
+    
+    #region Handle Credentials on basic authentication
+    $UserName = ''
+    $Password = ''
+    
+    if ($Authentication -like "Basic")
```

Comment:
Because I try not to use equal on two objects of different type. Too often equal fails when for most purposes it would should be correct from a users perspective.
Like compares the string representation. From the point of view of everyday usage it is less error prone.